src/components/ErrorBoundary/ErrorBoundary.tsx

```diff
@@ -3,23 +3,23 @@ import { getLogger } from "../../utils/logger";
 
 const { stack } = getLogger("component:errorBoundary");
 
-interface ErrorBoundaryState {
-  hasError: boolean;
-  error?: Error;
-}
-
-export interface ErrorBoundaryRendererProps {
+export interface FallbackComponentProps {
   error?: Error;
   recover: () => void;
 }
 
+export type FallbackComponentType = FunctionComponent<FallbackComponentProps>;
+
 export interface ErrorBoundaryProps {
-  renderer: ErrorBoundaryRenderer;
-  onError?: (error: Error) => void;
```

Comment:
question(non-blocking): Will there be situations where we need to do something specifically when there is a different error boundary? Like, when an error occurs, maybe `CertificateViewerErrorBoundary` wants to send error to ServiceA but `AnotherErrorBoundary` wants to send error to ServiceB? I think I probably was thinking there might be such cases and added the `onError` callback. But so far still no use case for it though and so removing it is actually fine too.

Reply:
oh i didn't thought that far yet. i had the impression we can create another separate error boundary for another potential problematic component if need

Reply:
Oh I see. Yea, creating a separate error boundary could work too. Just to clarify, this means that instead of reusing the `ErrorBoundary` component like with `CertificateViewerErrorBoundary`, when there is another separate error boundary like `AnotherErrorBoundary` and if it wants to have its own special error logic, we will reimplement its own `componentDidCatch`, is it like that?

Reply:
um i want to reuse `ErrorBoundary` ideally, how do i do that? did i just took away the reusability? 😬

on the other hand i might not use this react class based `ErrorBoundary` anymore. i saw sentry provides their own react [compatible one](https://docs.sentry.io/platforms/javascript/guides/react/components/errorboundary/#options) (looks same same but different). might still need to figure it out later

Reply:
Ohh.... If we are going to not use the `ErrorBoundary` anymore in future then it probably doesn't matter already. haha!

I didn't know Sentry provides their own `ErrorBoundary` too! I was thinking we have to use our `ErrorBoundary` and send to them. Nice! Looks like we could use theirs if they everything done for us already!😁